lib/rubocop/cop/rspec/nested_groups.rb

```diff
@@ -97,27 +97,26 @@ class NestedGroups < Cop
           "Configuration key `#{DEPRECATED_MAX_KEY}` for #{cop_name} is " \
           'deprecated in favor of `Max`. Please use that instead.'
 
-        def_node_search :find_contexts, ExampleGroups::ALL.block_pattern
-
         def on_top_level_describe(node, _args)
-          find_nested_contexts(node.parent) do |context, nesting|
+          find_nested_example_groups(node.parent) do |example_group, nesting|
             self.max = nesting
             add_offense(
-              context.send_node,
+              example_group.send_node,
               message: message(nesting)
             )
           end
         end
 
         private
 
-        def find_nested_contexts(node, nesting: 1, &block)
-          find_contexts(node) do |nested_context|
-            yield(nested_context, nesting) if nesting > max_nesting
+        def find_nested_example_groups(node, nesting: 1, &block)
+          example_group = example_group?(node)
+          yield node, nesting if example_group && nesting > max_nesting
```

Comment:
Couldn’t we optimise a bit by doing an early return if `nesting <= max_nesting`?

Reply:
Hmm. Not sure. Early return in this case means that none of child nodes will be processed and checked.

Reply:
Yeah, that’s probably not what we want to happen. :-)